src/compiler/bundle/typescript-plugin.ts

```diff
@@ -4,11 +4,10 @@ import { getModule } from '../transpile/transpiled-module';
 import { isString, normalizeFsPath } from '@utils';
 import type { Plugin } from 'rollup';
 import { tsResolveModuleName } from '../sys/typescript/typescript-resolve-module';
-import { isAbsolute } from 'path';
+import { isAbsolute, basename } from 'path';
 import ts from 'typescript';
 
-export const typescriptPlugin = (compilerCtx: d.CompilerCtx, bundleOpts: BundleOptions): Plugin => {
-  const tsPrinter = ts.createPrinter();
+export const typescriptPlugin = (compilerCtx: d.CompilerCtx, bundleOpts: BundleOptions, config: d.Config): Plugin => {
```

Comment:
Added similar typings in https://github.com/johnjenkins/stencil/pull/1/commits/41e8c3597e07adbb9e3b873517b8ef8c8ecdad16